dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/SubmissionService.java

```diff
@@ -168,8 +169,8 @@ public UploadBitstreamRest buildUploadBitstream(ConfigurationService configurati
 
         for (ResourcePolicy rp : source.getResourcePolicies()) {
             if (ResourcePolicy.TYPE_CUSTOM.equals(rp.getRpType())) {
-                ResourcePolicyRest resourcePolicyRest = converter.toRest(rp, Projection.DEFAULT);
-                data.getAccessConditions().add(resourcePolicyRest);
+                UploadAccessConditionDTO uploadAccessCondition = createAccessConditionFromResourcePolicy(rp);
```

Comment:
Similarly, with this change, I'd like to ask if we can add JavaDocs to the `buildUploadBitstream()` method (they are currently missing).  Describing this method would really help clarify the purposes of both `UploadBitstreamRest` (which is is building) and `UploadAccessConditionDTO` (which it uses to pass along access conditions from the Bitstream's resource policies)